### PR TITLE
Creating question form

### PR DESCRIPTION
**Pull request by StinaSoile**
opens next pages when questions are answered correctly. Accepts only one specific form of right answer (32, no thirty two, etc). Button flashes green if right answer, and red if wrong answer.
Try with bat-comic, the other comic does not yet have questions at all.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -6,13 +6,13 @@ export interface Page {
...
-  answer: string;
+  answer?: string;
 }
 
-export interface PageWithNoAnswer {
-  key?: string;
-  pictureName: string;
-  questionList?: QuestionWithNoAnswer[];
-}
+// export interface PageWithNoAnswer {
```
Comment (sepi4) [inline src/types.ts]: remove commented code

File: src/components/ComicBook/ComicBook.tsx [hunk 2]
```diff
@@ -3,23 +3,33 @@ import "./ComicPage.css";
...
+          console.log(error.response);
+        } else {
+          console.error(error);
+        }
+      }
     };
-    setKey(undefined);
-    getPages(comicName, key);
+    getPages(comicName, stateKey);
+    console.log(stateKey);
```
Comment (sepi4) [inline src/components/ComicBook/ComicBook.tsx]: remove console.log

File: src/components/ComicBook/ComicPage.css [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+  margin: 0.5rem;
+}
+
+.questionform {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+}
+
+/* .all-questions-container {
```
Comment (sepi4) [inline src/components/ComicBook/ComicPage.css]: commets

File: src/components/ComicBook/MathQuestions.tsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+  const [questions, setQuestions] = useState<Question[]>([]);
+  const [answers, setAnswers] = useState<string[]>([""]);
+  const [buttonClasses, setButtonClasses] = useState("answer-button");
+
+  useEffect(() => {
+    let newQuestions: Question[] = [];
+    if (comic[page].questionList) {
+      const l = comic[page].questionList.length;
+      for (let q = 0; q < l; q++) {
+        const question = comic[page].questionList[q];
```
Comment (sepi4) [inline src/components/ComicBook/MathQuestions.tsx]: do not use `for` loop with indexing, use `for of`, `.map` or `.forEach`. Use normal `for` when you need to use indexing.

File: src/components/ComicBook/MathQuestions.tsx [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+  stateKey,
+  setKey,
+  comicName,
+  comic,
+  page,
+}: {
+  stateKey: string | undefined;
+  setKey: React.Dispatch<React.SetStateAction<string | undefined>>;
+  comicName: string;
+  comic: Page[];
```
Comment (sepi4) [inline src/components/ComicBook/MathQuestions.tsx]: Are you sure you need to send comic and page number, and then index page from comic? Is it possible to give one Page as a prop. Page should have questions in it?

File: src/components/ComicBook/MathQuestions.tsx [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+      if (answers[i].length <= 0) isAnswers = false;
+    }
+    return isAnswers;
+  };
+
+  const flashRed = () => {
+    setButtonClasses("answer-button wrong-answer");
+    setTimeout(() => {
+      setButtonClasses("answer-button");
+    }, 500);
```
Comment (sepi4) [inline src/components/ComicBook/MathQuestions.tsx]: this is too short for me at least

File: src/components/ComicBook/MathQuestions.tsx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+      setAnswers(new Array(l).fill(""));
+    } else {
+      setQuestions(newQuestions);
+    }
+  }, [page, comic]);
+
+  const handleAnswer = async (e: SyntheticEvent) => {
+    e.preventDefault();
+    if (
+      comic[page].questionList &&
```
Comment (sepi4) [inline src/components/ComicBook/MathQuestions.tsx]: Maybe this `if (...)` can be cleand? To own function 
```
if(thereAreAnswers(comic[page]) {
...
```

**Review by sepi4 — CHANGES_REQUESTED** [review]

**Comment by sepi4** [comment]
try using conventional commits 
https://gist.github.com/qoomon/5dfcdf8eec66a051ecd85625518cfd13

File: src/components/ComicBook/MathQuestions.tsx (hunk 7 above)
Comment (StinaSoile) [reply src/components/ComicBook/MathQuestions.tsx]: typescript says that questionList might be undefined, if I move these to a separate function. I did that, though. Now it just checks in two places if the same questionList exists: 
in that if(...), and in the function thereAreAnswers().

File: src/components/ComicBook/MathQuestions.tsx (hunk 6 above)
Comment (StinaSoile) [reply src/components/ComicBook/MathQuestions.tsx]: Changed to 1 sec. Still too short?

Comment (sepi4) [reply src/components/ComicBook/MathQuestions.tsx]: better

**Comment by sepi4** [comment]
Always have short heading in commit messages. Add more text if need on different lines
image

**Comment by sepi4** [comment]
You can try reword your commit messages and squash you commit (if you dare 👹).

**Review by sepi4 — APPROVED** [review]

**Pull request merged**